### PR TITLE
Some clarifications in preparation to automatically order linker depe…

### PR DESCRIPTION
**Pull request by mathjeff**
…ndencies

Test: Browse the code and determine whether it's easier to understand
Bug: 66260943
Change-Id: I9ee8c1d5461008b489b052f23ca78096d0089fd8

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -1227,6 +1227,7 @@ func (c *Context) findMatchingVariant(module *moduleInfo, possible []*moduleInfo
 }
 
 func (c *Context) addDependency(module *moduleInfo, tag DependencyTag, depName string) []error {
+
```
Comment (colincross) [inline context.go]: I usually only insert an empty line at the top of the function if the function signature wraps to multiple lines, since gofmt will indent the wrapped signature the same as the first statement.

File: context.go [hunk 2]
```diff
@@ -3208,7 +3208,7 @@ func (c *Context) writeLocalBuildActions(nw *ninjaWriter,
 	return nil
 }
 
-func beforeInModuleList(a, b *moduleInfo, list []*moduleInfo) bool {
+func aPrecedesBInModuleList(a, b *moduleInfo, list []*moduleInfo) bool {
```
Comment (colincross) [inline context.go]: I agree the original is hard to parse, but I don't think this helps because any callers don't have an "a" and a "b", those are local to the function.  Maybe just add godoc that describes that this checks if the first argument is before the second argument in the list?

File: module_ctx.go [hunk 3]
```diff
@@ -278,6 +282,7 @@ type moduleContext struct {
 	handledMissingDeps bool
 }
 
+// TODO: expain how this differs from logicModule.Name() : does this relativize/contextualize paths?
```
Comment (colincross) [inline module_ctx.go]: See the comment on Module.Name().  logicModule.Name() is called exactly once, and returns what the module is initially know as.  moduleInfo.Name() returns the current name, which is initially the value of logicModule.Name(), but could be renamed by MutatorContext.Rename().

File: context.go [hunk 4]
```diff
@@ -960,7 +960,7 @@ func getStringFromScope(scope *parser.Scope, v string) (string, scanner.Position
 }
 
 // Clones a build logic module by calling the factory method for its module type, and then cloning
-// property values.  Any values stored in the module object that are not stored in properties
+// property values.  Any values stored in the module object that are not stored in Properties
```
Comment (colincross) [inline context.go]: I don't think Properties should be capitalized, that makes me think it is a Go type, but we are using it to refer to a class of structs.

File: module_ctx.go [hunk 5]
```diff
@@ -560,17 +565,17 @@ type BottomUpMutatorContext interface {
 // after parsing all Blueprint files, but before generating any build rules,
 // and is always called on dependencies before being called on the depending module.
 //
-// The Mutator function should only modify members of properties structs, and not
+// The Mutator function should only modify members of Properties structs, and not
```
Comment (colincross) [inline module_ctx.go]: I don't think this should be capitalized

File: module_ctx.go [hunk 6]
```diff
@@ -679,10 +684,10 @@ func (mctx *mutatorContext) AddReverseDependency(module Module, tag DependencyTa
 	})
 }
 
-// AddVariationDependencies adds deps as dependencies of the current module, but uses the variations
-// argument to select which variant of the dependency to use.  A variant of the dependency must
-// exist that matches the all of the non-local variations of the current module, plus the variations
-// argument.
+// AddVariationDependencies adds deps as dependencies of the current module, but uses the
+// variations argument to select which variant of the dependency to use.  The dependency must
+// provide a a variant whose Variation coordinates match each non-local Variation coordinate of the current module and each coordinate
```
Comment (colincross) [inline module_ctx.go]: I don't understand the use of the word coordinates here.

File: package_ctx.go [hunk 7]
```diff
@@ -39,9 +39,13 @@ import (
 //     var (
 //         pctx = NewPackageContext("path/to/blah")
 //
+//         // Should appear in the generated ninja file as "g.path.to.blah.myPrivateVar = abcdef"
 //         myPrivateVar = pctx.StaticVariable("myPrivateVar", "abcdef")
+//
+//         // Should appear in the generated ninja file as "g.path.to.blah.MyExportedVar = ${myPrivateVar} 123456!"
```
Comment (colincross) [inline package_ctx.go]: ${g.path.to.blah.myPrivateVar}

File: package_ctx.go [hunk 8]
```diff
@@ -39,9 +39,13 @@ import (
 //     var (
 //         pctx = NewPackageContext("path/to/blah")
 //
+//         // Should appear in the generated ninja file as "g.path.to.blah.myPrivateVar = abcdef"
```
Comment (colincross) [inline package_ctx.go]: Maybe not worth mentioning here, but live_tracker.go strips the variables and rules down to ones that are referenced through a build statement.

**Comment by mathjeff** [comment]
never figured out a wording that we all liked (similar to "coordinates") to make extra obvious the distinction between "variation" and "variant"
